From d8e63a47735fbc5a6c569be6f847b8bda4ed78d2 Mon Sep 17 00:00:00 2001
From: Yuriy Umanets <yuriy_umanets@xyratex.com>
Date: Tue, 7 Jun 2011 09:51:34 -0600
Subject: [PATCH 109/177] - fixed bug with c2_cob_put(NULL) in case of aborted
 transaction; - fixed bug with not NULL fom for restarted transaction.

---
 mdservice/md_foms.c | 2 +-
 mdstore/mdstore.c   | 3 +--
 reqh/reqh.c         | 3 ++-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/mdservice/md_foms.c b/mdservice/md_foms.c
index 5a6f12f..cb95750 100644
--- a/mdservice/md_foms.c
+++ b/mdservice/md_foms.c
@@ -532,8 +532,8 @@ static int c2_md_rename_fom_state(struct c2_fom *fom)
                         rc = c2_md_rename(site->s_mdstore, &pfid_tgt, &pfid_src, 
                                           &tfid_tgt, &tfid_src, &tattr, &sattr,
                                           scob, &ctx->fc_tx->tx_dbtx);
+                        c2_cob_put(scob);
                 }
-                c2_cob_put(scob);
         } else {
                 rc = c2_md_store_locate(site->s_mdstore, &tfid_src, &scob, 
                                         C2_MD_LOCATE_STORED,
diff --git a/mdstore/mdstore.c b/mdstore/mdstore.c
index ede43ce..2197279 100644
--- a/mdstore/mdstore.c
+++ b/mdstore/mdstore.c
@@ -776,8 +776,7 @@ restart:
                         break;
                 }
                 rc = c2_md_store_locate(md, &cob->co_nskey->cnk_pfid, 
-                                        &cob, C2_MD_LOCATE_STORED,
-                                        &tx);
+                                        &cob, C2_MD_LOCATE_STORED, &tx);
         }
 out:
         if (rc) {
diff --git a/reqh/reqh.c b/reqh/reqh.c
index d2aa4ba..3729c75 100644
--- a/reqh/reqh.c
+++ b/reqh/reqh.c
@@ -42,10 +42,11 @@ int c2_reqh_fop_handle(struct c2_reqh *reqh, struct c2_fop *fop, void *cookie)
 	struct c2_fop_ctx    ctx;
         struct c2_fop_env    env;
         struct c2_dtx        tx;
-	struct c2_fom	    *fom = NULL;
+	struct c2_fom	    *fom;
 	int                  result;
 
 restart:
+        fom = NULL;
         C2_SET0(&ctx);
         C2_SET0(&env);
 
-- 
1.8.3.2

